Fix validity check for IJK+ coordinates

saprykin opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/h3?pullRequest=271) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/h3?pullRequest=271) before we can accept your contribution.<br/><hr/>**Aleksandr Saprykin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/h3?pullRequest=271) it.</sub>

isaacbrodsky commented:
Thank you for catching this! Would you mind adding a test case to demonstrate this is fixed?

saprykin commented:
I added additional test cases when j and k components of IJK coordinates are out of range. With previous version new tests are not failing though they must. Hopefully I added it to right place.
